lib/delete.js

```diff
@@ -23,14 +23,21 @@ module.exports = function(bucket, options) {
     };
 
     function removed(err) {
-      if (err && err.statusCode !== 404) return callback(awsError(err, params));
+      if (err && err.name !== 'NoSuchKey') return callback(awsError(err, params));
```

Comment:
I feel like 404 would capture more than just NoSuchKey, can we add `NoSuchBucket` too?